### PR TITLE
feat: add presentation components for schedulelist and schedules to app state slice

### PR DESCRIPTION
**Pull request by tarnas14**
Description
---

- [x] add scheduling modal to mining view
- [x] display a list of schedules (hardcoded in initial state for now)
- [x] handle enabling/disabling schedules
- [x] handle schedule select and removal on 'backspace' or 'delete'
- [x] handle add/edit action
- [ ] form for creating/editing schedule - TODO (next PR #168 )
- [ ] persisting scheduling state between application starts - TODO (next next PR)
- [ ] mining scheduling based on schedule state - TODO (next next next PR)

known issues:
- non-serializable data in the redux state (`Date`) - fixed in #175)

Motivation and Context
---

#43 

How Has This Been Tested?
---

mining_scheduling

selection states & aligned
image
image

### CODE REVIEW

File: applications/launchpad/gui-react/src/containers/MiningContainer/Scheduling/ScheduleList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+          cancel={() => null}
+          addSchedule={addSchedule}
+          toggle={() => null}
+          edit={() => null}
+          remove={() => null}
+        />
+      </ThemeProvider>,
+    )
+
+    const addScheduleButton = screen.getByText('Add schedule')
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/containers/MiningContainer/Scheduling/ScheduleList.test.tsx:48]: NIT: I would rather avoid identifying an element with hard-coded text (`Add schedule`).

I see two alternatives:
a) use `data-testid` and `screen.getByTestId`
b) use locales (ie. `screen.getByText(t.mining.scheduling.add)`)

File: applications/launchpad/gui-react/src/types/general.ts [hunk 2]
```diff
@@ -5,6 +5,25 @@ export type CoinType = 'xtr' | ' xmr'
...
+}
+
+export type Schedule = {
+  id: string
+  enabled: boolean
+  days?: number[]
+  date?: Date
+  interval: Interval
+  type: MiningNodeType[]
+}
```
Comment (tomaszantas) [inline applications/launchpad/gui-react/src/types/general.ts:25]: Idea: _(sharing my thoughts that we can discuss later, not necessary any change request/suggestion)_

I'm wondering if there is anything else that may require scheduling. The only thing I've recognised is a periodical checking of new Docker images (and maybe availability of the new app version). However, it's something that can be also handled on the app launch, so it not necessary has to be scheduled.

Nevertheless, we may consider two improvements here:
1. having dedicated store/reducer for scheduling (instead having this in `app`)
2. adding field `scheduleType` or `eventType` that would define what action needs to be run. This way, we can schedule any event. (start mining, check images etc.).

**Review by tomaszantas — APPROVED** [review]
Looks good!
I left one _NIT_ to the test, and some thoughts in another.

File: applications/launchpad/gui-react/src/containers/MiningContainer/Scheduling/ScheduleList.test.tsx (hunk 1 above)
Comment (tarnas14) [reply applications/launchpad/gui-react/src/containers/MiningContainer/Scheduling/ScheduleList.test.tsx:48]: I'll use locale in next PR

File: applications/launchpad/gui-react/src/types/general.ts (hunk 2 above)
Comment (tarnas14) [reply applications/launchpad/gui-react/src/types/general.ts:25]: I haven't really thought about other schedules, in my mind periodical checks are a different beast - we would probably have a set period, configurable per build, so I don't think it's needed to have them stored as schedules in redux store

**Pull request merged**